notebooks/working/investigating_deconfounder.py

```diff
@@ -166,14 +174,22 @@
 results_df_full['true_params'] = coeffs_true
 results_df_full
 
-model_partial = smf.ols("y ~ a+b+c+d+e+f ", data=df)
+model_partial = smf.ols("y ~ a+b+c+d+e+f ", data=df) ## Ommit the confounder from OLS
 results_partial = model_partial.fit()
 results_df_partial = pd.DataFrame(results_partial.params, columns = ['estimated_params'])
 results_df_partial['std_errors'] = results_partial.bse
 results_df_partial['true_params'] = coeffs_true[:-1]
 results_df_partial
 
 # ## Factor analysis: Is there evidence of latent factors?
+#
+# How do we go about checking for whether a confounder exists (in this example, we know that a confounder exist since we defined the data generating process, but in the real world, the data generating process is more often than now unknown to the analyst). 
+#
+# There are a multitude of tests and ways to go about answering this question. A simple, off-the-shelf check is the bartlett sphericity test, which essentially just compares the correlation matrix of our covariates to the identity matrix. This test is used to check whether our covariate space can be reduced to a lower-dimensional space of factors that explain the variation in our data. 
```

Comment:
Exactly as you wrote, the bartlett sphericity test only tests for whether one's correlation matrix is different from the identity matrix.

While a latent confounder might lead to a non-identity correlation matrix, it need not always do so since not all dependencies are linear. For instance, the correlation can be zero even if two random variables are dependent on one another. See https://en.wikipedia.org/wiki/Distance_correlation for examples and references.

Moreover, you can have correlations between random variables even if there are no latent confounders.

As a result, I would not use this test to determine if we have latent confounding. If one fails the test, then we know there is correlation between the variables but not whether latent confounding is causing the correlation. If we pass the test, we still don't know that there is no latent confounding.